### PR TITLE
mgr/dashboard: fix empty tags 

### PR DESCRIPTION
**Pull request by Pegonzal**
Fixes: https://tracker.ceph.com/issues/64911

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test dashboard

**Review by nizamial09 — APPROVED** [review]
Once @afreen23 's comments are addressed i am good with this

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts [hunk 1]
```diff
@@ -452,6 +452,7 @@ export class RgwBucketFormComponent extends CdForm implements OnInit, AfterViewC
   }
 
   private tagsToXML(tags: Record<string, string>[]): string {
+    if (tags.length === 0) return null;
```
[the remark this answers is not in the document]
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts]: Now returning `''` for proper return type match

**Comment by Pegonzal** [comment]
Thanks for the reviews @afreen23 , @nizamial09.

I just added a variable `dirtyTags` so when editing the bucket if you remove all your tags, the bucket gets updated as well

**Comment by afreen23** [comment]
Retested with new changes:

### creating new bucket : shows no tagset field

```
{
    "bucket": "my-bucket-100",
    "tenant": "",
    "versioning": "off",
    "zonegroup": "02127924-b21e-4b80-a151-598973d0683f",
    "placement_rule": "default-placement",
    "explicit_placement": {
        "data_pool": "",
        "data_extra_pool": "",
        "index_pool": ""
    },
    "id": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.5",
    "marker": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.5",
    "index_type": "Normal",
    "index_generation": 0,
    "num_shards": 11,
    "object_lock_enabled": false,
    "mfa_enabled": false,
    "owner": "dashboard",
    "ver": "0#1,1#1,2#1,3#1,4#1,5#1,6#1,7#1,8#1,9#1,10#1",
    "master_ver": "0#0,1#0,2#0,3#0,4#0,5#0,6#0,7#0,8#0,9#0,10#0",
    "mtime": "2024-03-20T11:11:38.979076Z",
    "creation_time": "2024-03-20T11:11:38.938548Z",
    "max_marker": "0#,1#,2#,3#,4#,5#,6#,7#,8#,9#,10#",
    "usage": {},
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    }
}
```

### created a bucket with tags: shows tagset field with tags
```
{
    "bucket": "tags-bucket",
    "tenant": "",
    "versioning": "off",
    "zonegroup": "02127924-b21e-4b80-a151-598973d0683f",
    "placement_rule": "default-placement",
    "explicit_placement": {
        "data_pool": "",
        "data_extra_pool": "",
        "index_pool": ""
    },
    "id": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.6",
    "marker": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.6",
    "index_type": "Normal",
    "index_generation": 0,
    "num_shards": 11,
    "object_lock_enabled": false,
    "mfa_enabled": false,
    "owner": "afreen",
    "ver": "0#1,1#1,2#1,3#1,4#1,5#1,6#1,7#1,8#1,9#1,10#1",
    "master_ver": "0#0,1#0,2#0,3#0,4#0,5#0,6#0,7#0,8#0,9#0,10#0",
    "mtime": "2024-03-20T11:13:39.674943Z",
    "creation_time": "2024-03-20T11:13:39.629949Z",
    "max_marker": "0#,1#,2#,3#,4#,5#,6#,7#,8#,9#,10#",
    "usage": {},
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "tagset": {
        "foo": "bar"
    }
}
```

### removing all tags from a bucket with tags:  shows tagset field but with empty json

```
{
    "bucket": "tags-bucket",
    "tenant": "",
    "versioning": "off",
    "zonegroup": "02127924-b21e-4b80-a151-598973d0683f",
    "placement_rule": "default-placement",
    "explicit_placement": {
        "data_pool": "",
        "data_extra_pool": "",
        "index_pool": ""
    },
    "id": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.6",
    "marker": "3da6050a-9a39-4247-a32e-7be89ee62ba4.74841.6",
    "index_type": "Normal",
    "index_generation": 0,
    "num_shards": 11,
    "object_lock_enabled": false,
    "mfa_enabled": false,
    "owner": "afreen",
    "ver": "0#1,1#1,2#1,3#1,4#1,5#1,6#1,7#1,8#1,9#1,10#1",
    "master_ver": "0#0,1#0,2#0,3#0,4#0,5#0,6#0,7#0,8#0,9#0,10#0",
    "mtime": "2024-03-20T11:14:34.368753Z",
    "creation_time": "2024-03-20T11:13:39.629949Z",
    "max_marker": "0#,1#,2#,3#,4#,5#,6#,7#,8#,9#,10#",
    "usage": {},
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "tagset": {}
}
```

**Review by cloudbehl — APPROVED** [review]
LGTM

**Pull request merged**